### DISCUSSION TITLE
Text dynamically in params ?

### DISCUSSION THREAD

**Discussion by marlow-dev** [enesozturk/react-native-hold-menu#45] [Q&A]
Hello ! 😎

I would like to change the text of the parameters dynamically, for example I can like a post so there is the text which is equal to "like" but when the post is liked i would like there to be displayed " Not to like ", I tried a lot of solutions, but nothing very effective. Do you have any idea how I can do this? 

Thank you ! ❤️

**Comment by kayode0x** [comment]
I also have this problem. Have you fixed it yet?

**Comment by marlow-dev** [comment]
Hi, 

For me not yet, It's on to-do list, **unresolved** issue at this time

**Reply by kayode0x** [reply]
@enesozturk hi, any update on this?

**Reply by enesozturk** [reply]
The package currently supports this, if you change the list of items that you pass to `HoldMenu`, it will change the menu after opening it again. How did you try to get it? Could you please send code snippets?

**Reply by kayode0x** [reply]
Thanks for the reply @enesozturk, this is a mini setup I have. 
The problem is that the menu items don't update when the route changes. I tried using useMemo and usecallBack but they still don't work.

```
const menuItems = [
        {
            text: 'Open',
            icon: "ios-open-outline",
            onPress: openBookmark,
        },
        {
            text: 'Edit',
            icon: "ios-create-outline",
            onPress: editBookmark,
        },
        
        // NOW THIS PART DOESNT UPDATE WHEN THE ROUTE CHANGES.
        routeName === "Archive" 
        ?
        {
            text: 'Restore',
            icon: 'ios-reload-outline',
            onPress: restoreBookmark,
            withSeparator: true,
        },
        :
        {
            text: 'Archive',
            icon: 'ios-archive-outline',
            onPress: archiveBookmark,
            withSeparator: true,
        },
        {
            text: 'Delete',
            icon: "ios-trash-outline",
            isDestructive: true,
            onPress: askDelete,
        },
    ];
```

```
<HoldItem
            items={menuItems}
>
            <BookmarkListContainer
                onPress={() => viewLink(data.links)}
                activeOpacity={1}
            >
               {...rest}
            </BookmarkListContainer>
</HoldItem>
```

**Reply by enesozturk** [reply]
Did you try to give the list to `useMemo` or `useCallback` as a dependency?

**Reply by kayode0x** [reply]
sorry for the late response but i did,
